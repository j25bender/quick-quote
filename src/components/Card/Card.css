.all-cards {
  display: flex;
  flex-direction: row;
  overflow-y: scroll; }

.quote-container {
  display: flex;
  flex-direction: row; }

.next-button {
  position: absolute;
  background-image: url("../../assets/right-arrow3.png");
  background-size: 40px;
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-color: transparent;
  border: none;
  right: 30px;
  top: 50%;
  animation: slide 0.5s forwards; }

.favorite-button {
  background-image: url("../../assets/unfavorited.png");
  background-size: 40px;
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-color: transparent;
  border: none; }

.favorite-button:hover {
  background-image: url("../../assets/hover-favorite.png");
  background-size: 40px;
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-color: transparent;
  border: none; }
