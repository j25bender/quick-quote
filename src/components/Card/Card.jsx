import React from 'react';
import './Card.css'
import PropTypes from 'prop-types';

const Card = (props) => {
  const { quote, author, categories } = props.data;
  const cardHtml = <div>
                    <h4 className="quote">{ quote }</h4>
                    <h5 className="author">{ author }</h5>
                    <button className="favorite-button" onClick={ () => props.handleFavoriteClick({ props }) }></button>
                   </div>
  if(props) {
    const favoriteCheck = props.data.favorite ? "favorite" : "nonFavorite";
    const categoryList = categories !== undefined ? categories.map( (cat, i) => <p key={ i }>{ cat }</p> )                                             : <div className="no-categories"></div>;
    return (
      <div>
        <div className="quote-container">
          <div className={ favoriteCheck }>
            { cardHtml }
            <div className="categories">
              { categoryList }
            </div>
          </div>
          <button className="next-button" onClick={ () => props.handleClick(categories) }></button>
        </div>
      </div>
    )
  }
}

Card.propTypes = {
  data: PropTypes.shape({
    quote: PropTypes.string.isRequired,
    author: PropTypes.string,
    id: PropTypes.string.isRequired,
    categories: PropTypes.arrayOf(PropTypes.string)
  }).isRequired,

  handleClick: PropTypes.func.isRequired
};

export default Card;