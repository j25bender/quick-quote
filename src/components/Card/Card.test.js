import React from 'react';
import Card from './Card';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Card Component Test', () => {
  const mockHandleClick = jest.fn();
  const mockHandleFavoriteClick = jest.fn();
  const mockQuoteData = { quote: 'Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.', author: 'Robertson Davies', id: '65w3JLiRe5u_TKWtK5qyrweF', categories: Array(4) }

  it('Card should render quoteData correctly and match snapshot', () => {
    const wrapper = shallow(<Card key={ mockQuoteData.id } data={ mockQuoteData } handleClick={ mockHandleClick } handleFavoriteClick={ mockHandleFavoriteClick }/>);
    expect(wrapper).toMatchSnapshot();
  })
})