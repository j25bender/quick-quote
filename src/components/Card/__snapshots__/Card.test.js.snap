// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Test Card should render quoteData correctly and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "author": "Robertson Davies",
            "categories": Array [
              undefined,
              undefined,
              undefined,
              undefined,
            ],
            "id": "65w3JLiRe5u_TKWtK5qyrweF",
            "quote": "Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.",
          }
    }
    handleClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
                className="nonFavorite"
        >
                <div>
                        <h4
                                className="quote"
                        >
                                Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.
                        </h4>
                        <h5
                                className="author"
                        >
                                Robertson Davies
                        </h5>
                </div>
        </div>
        <div
                className="categories"
        >
                
        </div>
        <button
                onClick={[Function]}
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="nonFavorite"
>
            <div>
                        <h4
                                    className="quote"
                        >
                                    Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.
                        </h4>
                        <h5
                                    className="author"
                        >
                                    Robertson Davies
                        </h5>
            </div>
</div>,
          <div
            className="categories"
>
            
</div>,
          <button
            onClick={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <h4
                            className="quote"
              >
                            Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.
              </h4>
              <h5
                            className="author"
              >
                            Robertson Davies
              </h5>
</div>,
            "className": "nonFavorite",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4
                  className="quote"
>
                  Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.
</h4>,
                <h5
                  className="author"
>
                  Robertson Davies
</h5>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.",
                  "className": "quote",
                },
                "ref": null,
                "rendered": "Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Robertson Davies",
                  "className": "author",
                },
                "ref": null,
                "rendered": "Robertson Davies",
                "type": "h5",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
              undefined,
              undefined,
            ],
            "className": "categories",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="nonFavorite"
          >
                    <div>
                              <h4
                                        className="quote"
                              >
                                        Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.
                              </h4>
                              <h5
                                        className="author"
                              >
                                        Robertson Davies
                              </h5>
                    </div>
          </div>
          <div
                    className="categories"
          >
                    
          </div>
          <button
                    onClick={[Function]}
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="nonFavorite"
>
              <div>
                            <h4
                                          className="quote"
                            >
                                          Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.
                            </h4>
                            <h5
                                          className="author"
                            >
                                          Robertson Davies
                            </h5>
              </div>
</div>,
            <div
              className="categories"
>
              
</div>,
            <button
              onClick={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                <h4
                                className="quote"
                >
                                Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.
                </h4>
                <h5
                                className="author"
                >
                                Robertson Davies
                </h5>
</div>,
              "className": "nonFavorite",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4
                    className="quote"
>
                    Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.
</h4>,
                  <h5
                    className="author"
>
                    Robertson Davies
</h5>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.",
                    "className": "quote",
                  },
                  "ref": null,
                  "rendered": "Students today are a pretty solemn lot. One of the really notable achievements of the twentieth century has been to make the young old before their time.",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Robertson Davies",
                    "className": "author",
                  },
                  "ref": null,
                  "rendered": "Robertson Davies",
                  "type": "h5",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              "className": "categories",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
